utility/repoutil.py

```diff
@@ -103,30 +115,22 @@ def get_repo_owner(self) -> str:
         owner = os.path.basename(os.path.dirname(self.url))
         return owner.split(':')[-1]
 
-    def _clone(
-        self,
-        local_dir: t.Optional[str] = None,
-        config_username: t.Optional[str] = None,
-        config_user_email: t.Optional[str] = None,
-    ):
+    def _clone(self, clone_options: t.Optional[CloneOptions] = None):
         """
         Clone the specified repository and recursively clone submodules.
         This method raises a git.exec.GitCommandError if the repository does not exist.
 
         Arguments:
-            :param local_dir  (Optional[str]) Directory where to clone the repo.
-            By default, a new temporary directory is created.
-            :param config_username  (Optional[str]) Username to set in the git config.
-            :param config_user_email  (Optional[str]) Email to set in the git config.
+            :param clone_options  (Optional[CloneOptions]) Data for the repository cloning
         """
-        if local_dir:
+        if local_dir := clone_options.get('local_dir'):
             self.local_dir = local_dir
         else:
             self.local_dir = tempfile.mkdtemp()
         self.repo = Repo.clone_from(self.url, self.local_dir, multi_options=['--recurse-submodules'])
```

Comment:
I would leave this as something which is set by argument for example. We usually need to clone repo only because of IETF modules and rarely need to recursively pull all the submodules.
With submodules it takes too much space and time.